### PR TITLE
JCLOUDS-664: Change Azure Deploy representation

### PR DESCRIPTION
**Pull request by hsbhathiya**
Added OSVHD and DataVHD

Changes in Role

Added IOTypes

JCLOUDS-664:Azure Role representation

Role Instance handler with tests

JCLOUDS-664: Implemented Azure Role represenstaion

Added OSVHD and DataVHD

Changes in Role

Added IOTypes

JCLOUDS-664:Azure Role representation

Added VHD Handlers

Completed RoleHandler

JCLOUDS-664: Changes in Deployment implementation with test

Minor changes

### CODE REVIEW

**Comment by hsbhathiya** [comment]
@andreaturli  I made chnges to the azure deploy. can you review this?

File: azurecompute/src/main/java/org/jclouds/azurecompute/domain/RoleInstance.java [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+ */
+package org.jclouds.azurecompute.domain;
+
+import com.google.auto.value.AutoValue;
+import org.jclouds.javax.annotation.Nullable;
+
+import java.util.List;
+
+/**
+ * @see <a href="http://msdn.microsoft.com/en-us/library/azure/jj157193.aspx" >api</a>
```
Comment (andreaturli) [inline azurecompute/src/main/java/org/jclouds/azurecompute/domain/RoleInstance.java]: this points to Role rather than RoleInstance

File: azurecompute/src/main/java/org/jclouds/azurecompute/domain/Role.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+import java.net.URI;
+import java.util.List;
+
+import org.jclouds.javax.annotation.Nullable;
+
+import com.google.auto.value.AutoValue;
+
+/**
+ * @see <a href="http://msdn.microsoft.com/en-us/library/azure/jj157193.aspx" >api</a>
```
Comment (andreaturli) [inline azurecompute/src/main/java/org/jclouds/azurecompute/domain/Role.java]: shouldn't that contain a List<ConfigurationSet> ?

File: azurecompute/src/test/java/org/jclouds/azurecompute/features/OSImageApiLiveTest.java [hunk 3]
```diff
@@ -16,35 +16,24 @@
...
 import org.testng.annotations.BeforeClass;
 import org.testng.annotations.Test;
 
-import com.google.common.base.Function;
 import com.google.common.collect.ImmutableSet;
 
 @Test(groups = "live", testName = "OSImageApiLiveTest")
 public class OSImageApiLiveTest extends BaseAzureComputeApiLiveTest {
 
-   private ImmutableSet<String> locations;
```
Comment (andreaturli) [inline azurecompute/src/test/java/org/jclouds/azurecompute/features/OSImageApiLiveTest.java]: why did you remove this check from the test?

**Comment by andreaturli** [comment]
thanks for that @hsbhathiya!
Btw, it is a bit hard to review this PR all together, can you maybe split it into multiple PRs.
In fact, the code looks ok, but I'd like to have time to look at the official Azure documentation to see if the Value objects cover the spec correctly. Just to give you an example, `Role` doesn't contain `ConfigurationSet`.

For example a PR can contain:
- OSVirtualHardDisk
- OSVirtualHardDiskHandler
- OSVirtualHardDisksHandlerTest
- osvirtualharddisk.xml

so that the review process is quite straightforward?

**Comment by hsbhathiya** [comment]
@andreaturli  Thanks for quick response. I made https://github.com/jclouds/jclouds-labs/pull/125

**Comment by hsbhathiya** [comment]
This PR wil be splited into few smaller PRs
